### ISSUE TITLE
ISSUE (Windows): OFFLINE installation of dnscrypt-proxy Service is impossible (again)

### ISSUE THREAD

**Issue by iWARR** [DNSCrypt/dnscrypt-proxy#996]
**v2.0.29 Final**

If Network Adapter have no Internet connection, then:

- long pause
- service not installed, however it showing usual cmd-message:
```
Thank you for using DNSCrypt-Proxy!
Press [Enter] to exit...
```
The main goal I try to describe is to achieve **sure** service installation first (online or not) on Windows, that not depends on wether succeeded full scenario (that waiting for connection) or not.

As soon as Internet connection become available, then all necessary things (md-files) will be dowloaded after manual launching "service-restart.bat".

The bug was not fixed, OFFLINE service installation is still impossible on Windows, relative to issue https://github.com/DNSCrypt/dnscrypt-proxy/issues/977#issue-510498901 . It was not about `netprobe_timeout`

"service-install.bat" waiting for connection some time (it's normal), but if no connection, then service won't install (that's an issue).

**Comment by jedisct1** [comment]
The long pause depends on your netprobe setting. You can reduce it with `-netprobe-timeout 0`.

If you want to install the service with no connectivity, here's what you can do:

- Set `offline_mode = true`
- `dnscrypt-proxy -service install -netprobe-timeout 0`
- Set `offline_mode = false`
- `dnscrypt-proxy -service start`

**Comment by iWARR** [comment]
Further TESTS: **After reboot** the fresh service installation start working as expected (OFFLINE installation now really possible). So I'm happy the bug was fixed in final version anyway (that fix was very important).

And (luckily) none of proposed manipulations in the last post are necessary.

Sorry for inconvenience. Thank you!

**Comment by iWARR** [comment]
Sadly, the problem still exists. I'm stuck again installing new version offline (same as ten times before).

Some additional info for you:

- if no `md` files in the directory, then offine installation will fail.
- if I manually put `md` files inside the directory, then I can install dnscrypt-proxy in offline mode.

As an experiment, I put 2-days-old `md` files inside the freshy prepeared pack v2.0.31 and this worked. (I'm doubting are old `md`s a sec issue, or not...? I mean, invalid time marks).

Also (already having `md` files) restarting the service online is much faster then offline (none of other changes, just disabling network and comparing). Probably, this is expected behavior and not as big issue as offline installation fail.

Anyway, I think this is a not good behavior. Probably, for other people also.
You can't: or **prepare other machines** (install/fix/try), or **update** dnscrypt-proxy when offline. And expect flawless connection later, after Network become available - they power on, connect and go (I want to expect, that previously installed service on sertain machine will start automatically with my presets, that's it).

To summarize. If you can't install software offline, it's bad (especially on Windows).
Just take into account this small, but important facts.

Thank you for your attention.
